### PR TITLE
Fix `list_folders` to keep dots in the middle of the paths

### PR DESCRIPTION
**Pull request by grf53**
There was a bug in the `GcsBucket.list_folders` method, that all the dots(`"."`) in the folder paths were gone in its returned values.
That occurred by `.replace(".", "")` after getting paths. Maybe it is because there is unwanted `"."` if there is a file in the current position (refers to #121). We can use a simple filter, instead.

There was no tests with example blob under a folder named with dots. And there wasn't even the test for `list_folders` for the root. So I added along with the fix.

I am not sure including `""` as one of the result is wanted instead of `"."`. I thought it can be just omitted.

### Example
`bucket` is a `GcsBucket` instance that points a bucket with blobs with paths:
- `"blob.txt"`
- `"base_folder/nested_blob.txt"`
- `"dotted.folder/nested_blob.txt"`
#### Before
```
bucket.list_folders()  # ["", "base_folder", "dottedfolder"]
```

#### After
```
bucket.list_folders()  # ["base_folder", "dotted.folder"]
```

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-gcp/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-gcp/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by grf53** [comment]
@ahuang11 @razvanvilceanu Could you review this, kindly? You are the authors of #121, so I think you guys are the best reviewers of this.

**Comment by razvanvilceanu** [comment]
sure, can you assign it to me?

**Review by razvanvilceanu — APPROVED** [review]

**Comment by grf53** [comment]
@razvanvilceanu Thanks! I couldn't request review to you(blocked).

**Comment by grf53** [comment]
I am still blocked to proceed with message on this page:
```
Code owner review required
Waiting on code owner review from PrefectHQ/integrations.
This workflow requires approval from a maintainer.
```

@billpalombi Hello, I can see you approved some merged PRs that is required to get approvals of code owners.
If you are a code owner, then I'll be appreciated to you to review this.

**Comment by billpalombi** [comment]
Hi @grf53, I am not a code owner for this repo, but I will make sure that the right people are aware.

**Comment by grf53** [comment]
> I will make sure that the right people are aware.

Thank you so much!

File: CHANGELOG.md
Comment (desertaxle) [inline CHANGELOG.md:11]: Can you please add this line back?

**Review by desertaxle — COMMENTED** [review]
Thanks for contributing @grf53! This change looks good to me overall. I'm going to QA these changes against a real GCS bucket and I'll approve once I've confirmed everything is working as expected.

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Comment by grf53** [comment]
@desertaxle Grateful for your feedbacks! I reflected your suggestions. Please take a look.

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**
